jujugui/static/gui/src/app/app.js

```diff
@@ -303,7 +303,11 @@ YUI.add('juju-gui', function(Y) {
           if (spec.callback) { spec.callback.call(this, evt, target); }
           // HACK w/o context/view restriction but right direction
           if (spec.fire) {
-            this.views.environment.instance.topo.fire(spec.fire);
+            if (spec.fire.startsWith('topo.')) {
```

Comment:
If the fire starts with topo then dispatch a new event else fire it from the topo? 

Can you add a comment to explain this, as it seems backwards :)